### ISSUE TITLE
Add feature for downloading styles through vale.

### ISSUE THREAD

**Issue by adamtabrams** [vale-cli/vale#212]
The ability to download and update styles using vale would make the setup process much quicker and easier. It also would make the process more consistent across different operating systems. The desire behavior would look something like this:

`vale ls-styles`
```
Google - the Google Developer Documentation Style Guide
Joblint - test tech job posts for issues with sexism, culture, expectations, etc
Microsoft - the Microsoft Writing Style Guide
proselint - suggestions from the world’s greatest writers and editors
write-good - naive linter for English prose for developers who can't write good
CockroachDB - the CockroachDB style guide
```

Then you could install one to the styles path like this:
`vale install write-good`

And you could update the styles you've installed with something like this:
`vale update`

**Comment by omfgnuts** [comment]
Vale Server (https://github.com/errata-ai/vale-server) already does this.

**Comment by adamtabrams** [comment]
@omfgnuts Is there a reason why vale doesn't have it yet?

**Comment by jdkato** [comment]
The primary use case (from a development and support perspective) for Vale is CI environments (where desktop apps / remote services aren't great fits). This means that you should probably be installing Vale into an environment with a fixed configuration rather than having your configuration change on the fly and potentially break a build unexpectedly.

On the other hand, Vale Server is intended to address the needs of writers using a local environment where features like automatic updates make more sense.

In other words, I view Vale and Vale Server as complementary projects with different goals -- so, natrually, their feature sets are also going to be different.

**Comment by adamtabrams** [comment]
@jdkato That distinction seems inaccurate. Vale is a linter. For most people, there's no reason to run your linter as a server. I understand this for a corporate usecase which benefits from easier setup and more centralization. But for your average developer, student, etc, vale is a simpler and more sensible option.

It's sad to see this project is ignoring feedback and failing to make basic improvement to vale in order to persuade people to uses their paid alternative.

**Comment by eskwayrd** [comment]
@adamtabrams Vale is open source. You can make suggestions. You can contribute solutions. Nobody owes you anything, and being snarky could easily cause your feedback to be ignored.

From my experience, @jdkato provides above average to excellent support. I don't know @jdkato, but I've successfully been using Vale for about a year now, and its sad to see someone casting aspersions about his project and his development/support practices.

**Comment by ChrisChinchilla** [comment]
As a maintainer of one of the integration options for Vale (the VScode extension), I have thought about adding something like this ability, and potentially even an `init project` function to the Vale CLI version, but I haven't got around to it yet. Downloading the styles manually is not too hard in the meantime, so while I acknowledge that getting your project setup has a slight learning curve, we probably need to add some extra steps to the docs more than anything first.

And I concur with @eskwayrd , @jdkato has generally been extremely responsive to feedback, and likely makes minimal income from the server version, I genuinely think it's a design decision above all else.

**Comment by adamtabrams** [comment]
@eskwayrd I'm really glad that you've had a good experience with the project. I do hope that vale continues to improve. I also think it's important to share about negative experiences because there's a lot to be learned from them. I'm speaking as someone who was really excited about vale and set it up on my system. There were certain times during the setup when I realized that the process could be greatly improved (config being in $HOME, not generating a config file, and having to manually download a single directory from the repo of each style I wanted to use).

So, in order to help improve the project, I posted this feedback as issues #212 and #211. I was totally willing to implement them on my own if needed, but wanted to see if there were extra considerations to be made. In both cases I was met with immediate resistance to my feedback, and the issues were closed and labelled `won't fix` before any discussion could be had. 

> Vale is open source. You can make suggestions. You can contribute solutions.

This is partially true. I can make suggestions, and I did. But if I want to contribute solutions, those have to be merged by a project maintainer. If they aren't willing to discuss and issue, they aren't likely to merge a PR to fix it. That leaves me with maintaining my own fork of the project or just finding a different project.
